samples/photo_album/src/reducers/UploadedPhotosReducer.js

```diff
@@ -0,0 +1,20 @@
+const UploadedPhotosReducer = (uploadedPhotos = [], action) => {
+    switch (action.type) {
+        case 'UPDATE_UPLOADED_PHOTO': {
+            const newPhoto = action.uploadedPhoto;
+            const index = uploadedPhotos.findIndex(current => current.id === newPhoto.id);
+
+            return (index === -1) ?
+                [newPhoto, ...uploadedPhotos] :
+                uploadedPhotos.map(current => current.id === newPhoto.id ? {...current, ...newPhoto} : current);
```

Comment:
You already have the index of the changed item, why iterate again, and not change that index?